src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -260,6 +262,15 @@ public bool IsExpanded
                 AddGroupToGroupCommand.RaiseCanExecuteChanged();
                 RaisePropertyChanged(nameof(IsExpanded));
                 RedrawConnectors();
+                ReportNodesPosition();
```

Comment:
`node.ReportPosition` is an expensive operation when dealing with a lot of nodes under the same group.
Please check the performance with (there should be a .dyn you can use in this [jira taks](https://jira.autodesk.com/browse/DYN-4600)

Also take a look at what code paths might be calling the IsExpanded setter (to get a better idea what to test) - could be direct function calls or RaisePropertyChanges...

Reply:
It is for when the group is uncollapsed and the annotation inside it needs to report the position. Otherwise, that's what happens:
![annotationNode](https://user-images.githubusercontent.com/89042471/173682878-5a7b4557-b2f1-49f4-b614-69a3a128d449.gif)

Reply:
If there is no performance penalty in doing "node.ReportPosition" then it is fine.
The fix would be to make the Connectors redraw when the corresponding NodeModels(start or end) change their IsProxyPort flag
https://github.com/DynamoDS/Dynamo/blob/master/src/DynamoCoreWpf/ViewModels/Core/ConnectorViewModel.cs#L1209